Sources/PT.PM.Matching/Patterns/PatternIntLiteral.cs

```diff
@@ -18,19 +18,31 @@ public PatternIntLiteral(long value, TextSpan textSpan = default)
             Value = value;
         }
 
-        public override string ToString() => Value.ToString();
+        public override string ToString() => $"{Value}";
 
         protected override MatchContext Match(Ust ust, MatchContext context)
         {
             if (ust is IntLiteral intLiteral)
             {
-                return Value == intLiteral.Value ? context.AddMatch(intLiteral) : context.Fail();
+                return Value == intLiteral.Value
+                ? context.AddMatch(ust)
+                : context.Fail();
+            }
+            if (ust is LongLiteral longLiteral)
```

Comment:
But what about `BigIntLiteral`?

Reply:
`PatternBigIntLiterl` handle big numbers

Reply:
But actual `BigIntLiteral` might be small enough to match int literal.

Reply:
`BigIntLiteral` may not be small enough, since it was created specifically for processing numbers that are too large for ordinary long and int values. That is, if the number is small enough to fit in long, then it will be in `LongLiteral`

Reply:
What about languages where it's possible to explicitly specify BigInt decimals, like F#?
```fsharp
let v : bigint = 2I
```

Reply:
my bad, did not know about such f# features

Reply:
Yeah, F# is in no way unique in this regard.